DataDog/datadog-operator
# fix: support lease obj in  kubectl datadog clusteragent leader cmd

### What does this PR do?

Thanks to this commit, the kubectl-plugin can detect if the `Lease` resource
is supported in the current cluster and then retrieve the cluster-agent leader from the
`Lease` resource instead of the ConfigMap annotation.
### Motivation

In recent K8s cluster, the Lease resource replace the ConfigMap to store
the cluster-agent leader token.


### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

* Deploy a DatadogAgent name for example `datadog`. this deployment should have the cluster-agent enabled.
* Install the kubectl plugin (doc: https://github.com/DataDog/datadog-operator/blob/main/docs/kubectl-plugin.md). A release candidate version can be install thank to the command `kubectl krew install --manifest-url https://github.com/DataDog/datadog-operator/releases/download/v1.8.0-rc.1/datadog-plugin.yaml`
* Run the command `kubectl datadog cluster-agent leader datadog`
* the command should return the cluster-agent leader pod name 

```
➜   kubectl datadog clusteragent leader datadog-agent
The Pod name of the Cluster Agent is: datadog-cluster-agent-7575489654-rdrmm
```

### Checklist
